CMakeLists.txt

```diff
@@ -1199,35 +1202,102 @@ target_link_libraries (ripple_libs INTERFACE NIH::nudb)
 
 #[===================================================================[
    import protobuf (lib and compiler) and create a lib
-   from our proto message definitions. There is fallback
-   logic for windows that builds some version of the protobuf
-   core library from source and downloads the protoc compiler
-   if needed
+   from our proto message definitions. If the system protobuf
+   is not found, fallback on EP to download and build a version
+   from official source.
 #]===================================================================]
 
 if (static)
   set (Protobuf_USE_STATIC_LIBS ON)
 endif ()
 find_package (Protobuf)
-if (MSVC AND NOT TARGET protobuf::protoc)
-  find_program (my_protoc NAMES protoc)
-  if (NOT my_protoc)
-    message (WARNING "No protoc found -- downloading now")
-    file (DOWNLOAD
-      "https://ripple.github.io/Downloads/protoc/2.5.1/protoc.exe"
-      ${CMAKE_BINARY_DIR}/tools/protoc.exe
-      STATUS status
-      TIMEOUT 30)
-    list (GET status 0 status_code)
-    list (GET status 1 status_string)
-    if (NOT status_code EQUAL 0)
-      message (FATAL_ERROR "Failed to download protoc: ${status_string}")
-    endif ()
-    set (my_protoc ${CMAKE_BINARY_DIR}/tools/protoc.exe)
+if (local_protobuf OR NOT TARGET protobuf::libprotobuf)
+  message (STATUS "using local protobuf build.")
+  if (WIN32)
+    # protobuf prepends lib even on windows
+    set (lib_pre "lib")
```

Comment:
nit: maybe protobuf_lib_pre because it's a specific quirk to protobuf

Reply:
good idea - changed here as well as in the soci EP